### DISCUSSION TITLE
Ability to change the sorting of results

### DISCUSSION THREAD

**Discussion by VinylVault** [joalla/discogs_client#110] [Ideas] [answered]
Hi team,

The Discogs API allows for setting the order of the results provided by a few metrics, `dateadded` being one example, is there a way to do this with this interface?

if so, can you please advise how?

if not, can it be added?

**Comment by AnssiAhola** [comment]
Hi @VinylVault 

You can sort search results by the `sort()` method
it accepts the sort key (ones supported by the API) and sort order
```python
results = client.search("Master Boot")
print(results[0])
>>> <Artist 5237488 'Master Boot Record'>
results.sort("dateadded", "asc")
print(results[0])
>>> <Release 19894672 'ZZ Top / Bill Wyman / Snowy White / Stevie Ray Vaughan - Blues Rock / The Best Of ZZ Top'>
```

**Reply by VinylVault** [reply]
Hey @AnssiAhola does this work at the collection or folder 0 level too?

**Reply by AnssiAhola** [reply]
It works only on API points that support pagination
On the clients side any object that inherits the [BasePaginatedResponse](https://github.com/joalla/discogs_client/blob/1003cd382c92cedcb971384d24f7c55de28209ad/discogs_client/models.py#L260) class.

So, unfortunately it does not work for the collection folders
However it does work if you use the `collection_items()` method in user object as that returns a `PaginatedList` object

```python
result = me.collection_items(release_id)
result.sort("dateadded", "asc") # PaginatedList
```

**Reply by VinylVault** [reply]
collection folders do support sorting ... can this please be added

https://www.discogs.com/developers#page:user-collection,header:user-collection-collection-items-by-folder

**Reply by AnssiAhola** [reply]
Hey @VinylVault
Do you mean sorting the items/releases in collection folder?
```python
collection_folder.releases.sort("dateadded", "asc")
```
If so then that _should_ be possible, haven't tested it myself though

PS. Sorry for the wait

**Reply by VinylVault** [reply]
Yes that kind of thing @AnssiAhola it should theoretically work whichever folder it's assigned on 0 or a custom folder

**Reply by AnssiAhola** [reply]
Ok just tested and it seems to work just fine
One thing to note that there's no such sort string as "dateadded" its just "added"
```python
user = client.user("username")

releases = user.collection_items(22155985)
# releases = user.collection_folders[0].releases  # works
# releases = user.collection_folders[1].releases  # works

releases.sort("added", "asc")
for release in releases:
    print(release.instance_id)
```

**Reply by AnssiAhola** [reply]
I think I originally got confused and thougth you meant sorting the actual folders like so
```python
user.collection_folders.sort("added", "asc")
```
which in hindsight doesn't make really any sense :smile:

**Comment by AnssiAhola** [comment, marked as answer]
```python
user = client.user("username")

releases = user.collection_folders[folder_id].releases 

releases.sort("added", "asc")
for release in releases:
    print(release.instance_id)
```
List of accepted sort keys for collection folders found [here](https://www.discogs.com/developers#page:user-collection,header:user-collection-collection-items-by-folder)

@VinylVault
I'll mark this as answered.
